Sources/ApolloCodegenLib/Templates/RenderingHelpers/GraphQLEnumValue+Rendered.swift

```diff
@@ -23,20 +23,64 @@ extension GraphQLEnumValue.Name {
     }
   }
 
+  /// Convert to `camelCase` from `snake_case`, `UpperCamelCase`, or `UPPERCASE`.
+  ///
+  /// 1. Returns a first lowercased string
+  /// 1. Capitalizes the word starting after each `_`
+  /// 2. Removes all inner `_`
+  /// 3. Preserves leading and trailing `_`
+  /// 4. Converts an all uppercase string into all lowercase
   private func convertToCamelCase(_ value: String) -> String {
-    if value.allSatisfy({ $0.isUppercase }) {
-      // For `UPPERCASE`. e.g) UPPER -> upper, STARWARS -> starwards
-      return value.lowercased()
+    // The source for this function is from the JSONDecoder implementation in Swift Foundation,
```

Comment:
@bignimbus @hwillson Is this in-line comment enough to be in compliance with the license? This is coming from Apple's open source swift foundation library. With modifications for our specific needed behaviors.

LGTM, just want to make sure this is the correct way to do this.

Reply:
I took another look at the license requirements and I'm now doubtful that statement is enough. Reading [this](https://fossa.com/blog/open-source-licenses-101-apache-license-2-0/#:~:text=Anyone%20who%20uses%20open%20source,made%20to%20the%20original%20code) we'd need to include a copy of the original copyright notice (not a problem) and a copy of the license itself (I don't want to add another license to the repo).

I'll look at rewriting the function tomorrow from scratch and we can get avoid this.

Reply:
This is done - https://github.com/apollographql/apollo-ios/pull/2745/commits/e6bc8a6bdf722da612721b975da7d720d9cbf6e0

Reply:
@AnthonyMDev you can take another pass at this, it's done - https://github.com/apollographql/apollo-ios/commit/e6bc8a6bdf722da612721b975da7d720d9cbf6e0.